cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/linkanalysis/LinkAnalysis.scala

```diff
@@ -0,0 +1,121 @@
+package com.twitter.cassovary.algorithms.linkanalysis
+
+import com.twitter.cassovary.graph.{Node, DirectedGraph}
+import com.twitter.cassovary.util.Progress
+import com.twitter.logging.Logger
+
+/**
+ * Base trait for link analysis algorithms.  The goal of these is to iteratively determine the importance
+ * of the nodes in the graph by analyzing how each of those nodes is connected to its neighbors.
+ * @tparam T `LinkAnalysis` must be generically typed by `Iteration` or one of its subclasses.  An `Iteration`
+ *          holds all of the pertinent information for a given algorithm.
+ */
+trait LinkAnalysis[T <: Iteration] extends (Option[T] => T) {
```

Comment:
The 'extends (Option[T] => T)' looks artificial. I'd remove it and let the user use run(). Also I'd merge the trait LinkAnalysis and abstract class AbstractLinkAnalaysis to one abstract class.